### ISSUE TITLE
[openid_user_client_role_protocol_mapper]  does not work for particular client (client_id_for_role_mappings)

### ISSUE THREAD

**Issue by semangard** [mrparkers/terraform-provider-keycloak#403]
Hello,

I have the following piece of code :

resource "keycloak_openid_user_client_role_protocol_mapper" "D_to_A_api1_client__user_A_api_1_roles_mapper" {
  realm_id  = module.collaborator_realm.id
  client_id = module.D_to_A_api1_client.id

  name = "A_api_1 User Roles"

  roles of each client stored separately.
  claim_name = "resource_access.a_api1.roles"
  client_id_for_role_mappings = module.A_api1_client.id   # BUG DOES NOT WORK
}

It generates the plan ::

 + resource "keycloak_openid_user_client_role_protocol_mapper" "D_to_A_api1_client__user_A_api_1_roles_mapper" {
      + add_to_access_token         = true
      + add_to_id_token             = true
      + add_to_userinfo             = true
      + claim_name                  = "resource_access.a_api1.roles"
      + claim_value_type            = "String"
      + client_id                   = "26ea7d07-f232-420b-89ab-8f1aa9d7ea30"
      + client_id_for_role_mappings = "2739fee5-e99e-41ab-ba58-5652472a8e77"
      + id                          = (known after apply)
      + multivalued                 = false
      + name                        = "A_api_1 User Roles"
      + realm_id                    = "collaborator2"
    }

BUT into the UI none client is selected into the list 'Client ID' (cf. screenshot)
![Keycloak Admin Console - Mozilla Firefox 2020-10-2](https://user-images.githubusercontent.com/57065741/96863049-968d6280-1466-11eb-8f2b-497c698c7ae0.png)

**Comment by tomrutsaert** [comment]
Documentation is wrong: 

Should be client_id instead of id

Try:

client_id_for_role_mappings = module.A_api1_client.client_id

When in doubt always also check the examples in:
https://github.com/mrparkers/terraform-provider-keycloak/tree/master/example

**Comment by tomrutsaert** [comment]
I am happy to merge quickly any PR improving the docs.

**Comment by semangard** [comment]
OK thanks it works now.

**Comment by mrparkers** [comment]
Good catch with the docs @tomrutsaert.  I really hate the discrepancy between `client.id` (the unique ID of the client in Keycloak) and `client_id` (the "client id" used within the various OIDC flows).  This is a good example of how easy it is to mix the two up.

**Comment by misteio** [comment]
Hey ! I have the following error when I fill client_id with both client client_id or with string  :
` error sending POST request to /auth/admin/realms/<realm-id>/clients/<client-id>/roles: 404 Not Found. Response body: {"error":"Could not find client"}`
I'm currently using Keycloak 12.0.4 + Terraform 0.14.7 + mrparkers/keycloak 2.3.0.

Has somebody same issue?
